### PR TITLE
[PPML] Add trusted-machine-learning

### PR DESCRIPTION
**Pull request by SEULiyao**
## Description

Add trusted-machine-learning

### 1. Why the change?

to provide a image used for machine learning

### 2. User API changes

add new scripts in `/scripts` that run ML example

### CODE REVIEW

File: ppml/trusted-machine-learning/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    git checkout branch-3.2.0-ppml && \
+    cd hadoop-common-project/hadoop-common && \
+    export MAVEN_OPTS="-Xmx2g -XX:ReservedCodeCacheSize=512m \
+        -Dhttp.proxyHost=$HTTP_PROXY_HOST \
+        -Dhttp.proxyPort=$HTTP_PROXY_PORT \
+        -Dhttps.proxyHost=$HTTPS_PROXY_HOST \
+        -Dhttps.proxyPort=$HTTPS_PROXY_PORT" && \
+    mvn -T 16 -DskipTests=true clean package && \
+    mv /opt/hadoop/hadoop-common-project/hadoop-common/target/hadoop-common-3.2.0.jar /opt/spark-${SPARK_VERSION}/jars/hadoop-common-3.2.0.jar
+# hive
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/Dockerfile:109]: we do not need hive which is a big-data tool

**Comment by Uxito-Ada** [comment]
{LR, GBT, random forest} × {local, k8s} × {native, sgx} examples should be appended later.

File: ppml/trusted-machine-learning/scripts/start-random-forest-classifier-on-local-native.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+cd /ppml
+
+/opt/jdk8/bin/java \
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/scripts/start-random-forest-classifier-on-local-native.sh:3]: where is input data come from?

Comment (Uxito-Ada) [reply ppml/trusted-machine-learning/scripts/start-random-forest-classifier-on-local-native.sh:3]: Need to add data-preparation code before spark running

File: ppml/trusted-machine-learning/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+ARG BASE_IMAGE_TAG=2.3.0-SNAPSHOT
+ARG SPARK_VERSION=3.1.3
+ARG TINI_VERSION=v0.18.0
+ARG JDK_VERSION=8u192
+ARG JDK_URL
+ARG SPARK_JAR_REPO_URL
+ARG SCALA_VERSION=2.12
+
+
+# Stage.1 Spark & Hadoop
```
Comment (glorysdj) [inline ppml/trusted-machine-learning/Dockerfile:11]: copy from trusted big data image directly?

Comment (SEULiyao) [reply ppml/trusted-machine-learning/Dockerfile:11]: It's based on trusted-bigdata Dockerfile. Some unnecessary content has been removed.

Comment (Uxito-Ada) [reply ppml/trusted-machine-learning/Dockerfile:11]: You can directly copy complied dependency from bigdata release like:

```
ARG BIGDATA_IMAGE_NAME=bigdl-ppml-trusted-bigdata-gramine
ARG BIGDATA_IMAGE_TAG=2.3.0-SNAPSHOT

FROM ${BIGDATA_IMAGE_NAME}:{BIGDATA_IMAGE_TAG}
......
```

@SEULiyao

File: ppml/trusted-machine-learning/log4j2.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/log4j2.xml:1]: where to add this file to image?

Comment (SEULiyao) [reply ppml/trusted-machine-learning/log4j2.xml:1]: > it was used to build the base image . not needed currently. i will delete it

**Comment by Uxito-Ada** [comment]
Too many scripts in one folder, can we refactor to as below:
scripts
--classification
----native
----sgx
--regression
----native
----sgx
--unsupervised
----native
----sgx

File: ppml/trusted-machine-learning/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ARG BIGDATA_IMAGE_TAG=2.3.0-SNAPSHOT
+ARG SPARK_VERSION=3.1.3
+ARG TINI_VERSION=v0.18.0
+ARG JDK_VERSION=8u192
+ARG JDK_URL
+ARG SPARK_JAR_REPO_URL
+ARG SCALA_VERSION=2.12
+
+# stage.1 bigdata image 
+FROM $BIGDATA_IMAGE_NAME:$BIGDATA_IMAGE_TAG as bigdata
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/Dockerfile:13]: remove this empty stage

**Comment by Uxito-Ada** [comment]
move to 7740

File: ppml/trusted-machine-learning/build-machine-learning-base-image.sh [hunk 6]
```diff
@@ -0,0 +1,47 @@
+export HTTP_PROXY_HOST=your_http_proxy_host
+export HTTP_PROXY_PORT=your_http_proxy_port
+export HTTPS_PROXY_HOST=your_https_proxy_host
+export HTTPS_PROXY_PORT=your_https_proxy_port
+export JDK_URL=http://10.239.45.10:8081/repository/raw/jdk/jdk-8u192-linux-x64.tar.gz
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/build-machine-learning-base-image.sh:5]: turn to http://your-http-url-to-download-jdk

File: ppml/trusted-machine-learning/build-machine-learning-base-image.sh [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+export SPARK_JAR_REPO_URL=http://10.239.45.10:8081/repository/raw/spark
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/build-machine-learning-base-image.sh:6]: remove internal url as well

File: ppml/trusted-machine-learning/build-machine-learning-base-image.sh [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+export LOCAL_IP=172.168.0.205
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/build-machine-learning-base-image.sh:7]: remove internal IP too

File: ppml/trusted-machine-learning/README.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+export NFS_INPUT_PATH=/YOUR_DIR/data
+export KEYS_PATH=/YOUR_DIR/keys
+export SECURE_PASSWORD_PATH=/YOUR_DIR/password
+export KUBECONFIG_PATH=/YOUR_DIR/kubeconfig
+export LOCAL_IP=$LOCAL_IP
+export DOCKER_IMAGE=YOUR_DOCKER_IMAGE
+sudo docker run -itd \
+    --privileged \
+    --net=host \
+    --name=machine-learning-graming \
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/README.md:121]: fix typo pls

**Comment by Uxito-Ada** [comment]
All of remaining comments above have been addressed in #7741

File: ppml/trusted-machine-learning/scripts/start-LDA-on-local-native.sh [hunk 10]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# source code link: https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/ml/LDAExample.scala
+cd /ppml
+
+input_file="data/mllib/sample_lda_libsvm_data.txt"
```
Comment (Uxito-Ada) [inline ppml/trusted-machine-learning/scripts/start-LDA-on-local-native.sh:5]: where is data/mllib created?

Comment (Uxito-Ada) [reply ppml/trusted-machine-learning/scripts/start-LDA-on-local-native.sh:5]: wget is only getting a file NOT creating the directory structure for you (mkdir -p /path/to/image/), you have to do this by urself

**Comment by Uxito-Ada** [comment]
Pls pay attention to code quality in future development @SEULiyao

**Pull request closed without merging**